Week3/HW3/3_moreJavascript/15_testEquality.js

```diff
@@ -0,0 +1,36 @@
+// x == y it should be true since it is not strict equality.
+// x === y  it should be false since it is strict equality.
+// z == y  that is true since it is defined like that.
+// z == x it should be true since it is not strict equality.
+
+
+function same(a,b){
+    var same = a.length === b.length,
+        result,
+        len;
+    if(!same) { result = false }
+    else {
+        // so far, they're the same
+        result = true;
+        len = a.length;
+        for(var i=0;result !== false && i < len; i++) {
+            if(a[i] !== b[i]) {
+                result = false;
+            }
+        }
+    }
+    
+    return result;
+};
+
+
+let x = [1, 2, 3];
+let y = [1, 2, 3];
+let z = y;
+
+same(x,y); 
+// prints: true
+same(z,y)
+// prints: true
+same(z,x)
```

Comment:
Aahaan. This isn't correct. 🚫 What was expected was to check for `x === y` , `x == y`, `z === y`, `z == y` and `z === x`.

In JavaScript, when we equate two objects(arrays being type of an object), the reference of those objects is checked for instead of their values. This is, however, not the case when we are dealing with primitive values such as `strings`, `numbers` or `boolean` because then JavaScript compare values instead of references. Let me know please if you still require further clarifications on this 🙂

Reply:
Thank you. I better understood now. Ok, i will do so.